xbmc/video/PlayerController.cpp

```diff
@@ -425,6 +428,31 @@ bool CPlayerController::OnAction(const CAction &action)
         return true;
       }
 
+      case ACTION_PLAYER_PROGRAM_SELECT:
+      {
+        std::vector<ProgramInfo> programs;
+        g_application.m_pPlayer->GetPrograms(programs);
+        CGUIDialogSelect *dialog = g_windowManager.GetWindow<CGUIDialogSelect>(WINDOW_DIALOG_SELECT);
+        if (dialog)
+        {
+          int playing = 0;
+          int idx = 0;
+          for (auto prog : programs)
+          {
+            dialog->Add(prog.name);
+            if (prog.playing)
+              playing = idx;
+            idx++;
+          }
+          dialog->SetHeading(CVariant{g_localizeStrings.Get(39109)});
+          dialog->SetSelected(playing);
+          dialog->Open();
+          idx = dialog->GetSelectedItem();
```

Comment:
this can be -1 if dialog is closed, better check against the bounds of `programs`

Reply:
right